packages/tenant-process/test/tenant.integration.test.ts

```diff
@@ -153,17 +157,26 @@ describe("Integration tests", () => {
           attributeId,
           updateVerifiedTenantAttributeSeed,
         });
-        const writtenEvent = await readLastEventByStreamId(
-          tenant.id,
-          postgresDB
-        );
+        const writtenEvent: StoredEvent | undefined =
+          await readLastEventByStreamId(
+            tenant.id,
+            eventStoreSchema.tenant,
+            postgresDB
+          );
+        if (!writtenEvent) {
+          fail("Creation fails: tenant not found in event-store");
+        }
+        expect(writtenEvent).toBeDefined();
         expect(writtenEvent.stream_id).toBe(tenant.id);
         expect(writtenEvent.version).toBe("1");
         expect(writtenEvent.type).toBe("TenantUpdated");
-        const writtenPayload = decodeProtobufPayload({
-          messageType: TenantUpdatedV1,
-          payload: writtenEvent.data,
-        });
+        const writtenPayload: TenantUpdatedV1 | undefined = protobufDecoder(
+          TenantUpdatedV1
+        ).parse(writtenEvent.data);
+
+        if (!writtenPayload) {
+          fail("impossible to decode TenantUpdatedV1 data");
+        }
```

Comment:
is this check generally done this way?

Reply:
I took inspiration from the agreement tests and this is how it was done